core/crypto/src/key_file.rs

```diff
@@ -11,6 +18,34 @@ pub struct KeyFile {
     pub account_id: String,
     pub public_key: PublicKey,
     pub secret_key: SecretKey,
+    #[serde(default)]
+    pub key_hash: KeyHash
+}
+
+fn secretKeyToVec(secret_key: SecretKey) -> Vec<u8> {
```

Comment:
There is no reason to not use strong types and coerce encrypted SecretKey to normal SecretKey.

Have two structures: `EncryptedKeyFile` and `KeyFile`. `KeyFile` is what we used to have before and pass around in the system. `EncryptedKeyFile` has the following fields and we only use it to deserialize/serialize from the disk and convert to `KeyFile`.

```
pub struct EncryptedKeyFile {
    pub account_id: String,
    pub public_key: PublicKey,
    pub encrypted_secret_key: EncryptedSecretKey,
    pub key_hashL KeyHash,
}
```
Then we also do not need to use `#[serde(default)]`.